syntax = "proto3";

message StockFeed {
  reserved 1 to 8;
  StockData stock_data = 9;
}

message StockData {
  string symbol = 1;
  double lastprice = 2;
  double volume = 3;
  double high = 4;
  double low = 5;
  double open = 6;
  double frequency = 7;
  double foreignbuy = 8;
  double foreignsell = 9;
  string date = 10;
  string timestamp = 11;
  double close = 12;
  double previous = 13;
  double value = 14;
  PriceChange change = 15;
  string order_verb = 16;
  int64 quantity = 17;
  bool is_index = 18;
  int64 sequence_number = 19;
  int64 order_book_id = 20;
  int64 order_number = 21;
  int64 match_number = 22;
  string itch_incoming_time = 26;
}

message PriceChange {
  double value = 1;
  double percentage = 2;
}